src/audio/smart_amp/smart_amp.c

```diff
@@ -111,8 +111,11 @@ static ssize_t smart_amp_alloc_mod_memblk(struct smart_amp_data *sad,
 
 	/* query the required size from inner model. */
 	ret = mod->mod_ops->query_memblk_size(mod, blk);
-	if (ret <= 0)
+	if (ret <= 0) {
+		if (ret == 0)
+			ret = -EINVAL;
```

Comment:
I'll propose to return early without error
```
if (ret < 0)
    goto error;

/* no memory requirement */ 
if (ret == 0)
    return 0;

/* allocate the memory block when returned size > 0. */
...
```

Reply:
We should also call this out in an inline comment that 0 here is an error code from smart amp (otherwise people will try and fix)

Reply:
I think what @johnylin76 is saying is that its not an error and was incorrectly treated as it if it was an error. I believe 0 is "not needed" and shouldnt have been triggering the error path.